### DISCUSSION TITLE
How to implement OAuth 2 - Password flow ?

### DISCUSSION THREAD

**Discussion by vimalraj-a** [panva/openid-client#341] [Q&A]
We are not able make authorization call from this library since we get the authorization url from client.

How do we send user credentials and custom headers for authorization? Below is auth flow detailed documentation that we are looking at.
https://www.ibm.com/support/knowledgecenter/SSMNED_5.0.0/com.ibm.apic.toolkit.doc/tutorial_apionprem_security_OAuth.html#OAuth_Tute__Choosing_Scheme

![image](https://user-images.githubusercontent.com/13977942/110363036-0e4fd880-8068-11eb-9c5f-1dc4791b1144.png)

**Comment by panva** [comment]
Use [`client.grant(body)`](https://github.com/panva/node-openid-client/blob/v4.5.1/docs/README.md#clientgrantbody-extras)

**Reply by vimalraj-a** [reply]
You mean, Instead of `client.authorizationURL()` call `client.grant()` and then callback will be hit by OAuth2 Server? @panva
